pkg/network/server.go

```diff
@@ -1442,22 +1442,15 @@ func (s *Server) tryInitStateSync() {
 
 	var peersNumber int
 	s.lock.RLock()
-	heights := make([]uint32, 0)
+	heights := make([]uint32, 0, len(s.peers))
 	for p := range s.peers {
 		if p.Handshaked() {
 			peersNumber++
-			peerLastBlock := p.LastBlockIndex()
-			i := sort.Search(len(heights), func(i int) bool {
-				return heights[i] >= peerLastBlock
-			})
-			heights = append(heights, peerLastBlock)
-			if i != len(heights)-1 {
-				copy(heights[i+1:], heights[i:])
-				heights[i] = peerLastBlock
-			}
+			heights = append(heights, p.LastBlockIndex())
 		}
 	}
 	s.lock.RUnlock()
+	slices.Sort(heights)
 	if peersNumber >= s.MinPeers && len(heights) > 0 {
```

Comment:
Something isn't clear here. Why do we need `peersNumber` variable then if it matches `len(heights)`?

Reply:
Fair enough.